server and agent env vars; storage classes; expose

mantis-toboggan-md opened the pull request:
EDIT: I've updated the PR to include #23 

#18 
![Screenshot 2025-06-17 at 3 03 04 PM](https://github.com/user-attachments/assets/ec09751e-77ff-42e3-84a9-b41561e96089)


#14

https://github.com/user-attachments/assets/fd9f936f-7f8a-4fc6-8fa4-2dd642a14a11



#12 - I also removed 'local path' as default storage class: instead, we'll look for a class with an annotation denoting that its the default. Note that a cluster may have storage classes, but no default - the UI will use 'Ephemeral' as the default value

![Screenshot 2025-06-17 at 3 02 23 PM](https://github.com/user-attachments/assets/8b43bfdd-9426-401a-80da-d2159dd3abfc)


I've also added an additional filter to the host cluster list to exclude clusters that are still provisioning (using the management cluster `isReady` property which is also used to gate access to cluster explorer)

eva-vashkevich commented on pkg/virtual-clusters/l10n/en-us.yaml:
```diff
@@ -6,16 +6,25 @@ k3k:
+      label: Number of server pods
+    envVars:
+      title: Server pod environment variables
+      keyLabel: Env variable
```
Environment variable

eva-vashkevich reviewed (changes requested):
I am getting an error when messing around with env variables:
https://github.com/user-attachments/assets/8d42a072-0c8f-4043-9813-c4b9f30ccf6b
I am also getting "Error from server (BadRequest): error when creating "./management-state/tmp/yaml-1158107077": Cluster in version "v1alpha1" cannot be handled as a Cluster: strict decoding error: unknown field "spec.expose.loadBalancer"" when trying to set load balancer. did it work for you? am I missing something?
Storage looks good tho

eva-vashkevich commented on pkg/virtual-clusters/components/CruK3KCluster/Networking.vue:
```diff
@@ -0,0 +1,327 @@
+
+  data() {
+    let exposeMode;
+    const exposeModes = {
```
why is it in data?

mantis-toboggan-md commented:
> I am getting an error when messing around with env variables: https://github.com/user-attachments/assets/8d42a072-0c8f-4043-9813-c4b9f30ccf6b I am also getting "Error from server (BadRequest): error when creating "./management-state/tmp/yaml-1158107077": Cluster in version "v1alpha1" cannot be handled as a Cluster: strict decoding error: unknown field "spec.expose.loadBalancer"" when trying to set load balancer. did it work for you? am I missing something? Storage looks good tho

I fixed env vars and the load balancer inputs. When I was looking at the k3k cluster CRD documentation I read the docs for args rather than env vars :L. Though JP said serverArgs would be good to have too - I've added them as well as addressed some other issues raised over slack:

- `serverEnvs` and `agentEnvs` should be an array of objects with name and value properties
- when the storage class is ephemeral the persistence type needs to be `ephemeral` , otherwise, the persistence type will be `dynamic`
- `storageRequestSize`  should default to 5Gi, not 1Gi and should also have an input
- TLS Subject Alternative Names should be validated as required when expose mode is ingress

momesgin reviewed (approved):
tested and lgtm
